lib/NativeCall/Types.rakumod

```diff
@@ -244,6 +244,10 @@ our class CArray is repr('CArray') is array_type(Pointer) {
             nqp::create(self)
         }
     }
+    multi method new(Str:D $str, :enc(:$encoding) = 'utf8', *%encode-opts) {
+        # Add NUL terminator to the encoded C string:
+        self.new($str.encode($encoding, |%encode-opts).list, 0)
```

Comment:
The problem with this
self.new($data.encode($encoding).list, 0)

is that at least one encoding, UTF-16, needs two 0 bytes to terminate the string.

Would this work?

self.new(($data~"\0").encode($encoding).list)

(also posted this same comment in the issue discussion)